heron/tmaster/src/cpp/manager/tmaster.cpp

```diff
@@ -142,6 +142,14 @@ void TMaster::EstablishTMaster(EventLoop::Status) {
   auto cb = [this](proto::system::StatusCode code) { this->SetTMasterLocationDone(code); };
 
   state_mgr_->SetTMasterLocation(*tmaster_location_, std::move(cb));
+
+  // if zk lost the tmaster location, tmaster quits to bail out and re-establish its location
+  auto cb2 = [this]() {
+    LOG(ERROR) << " lost tmaster location in zk state manager. Bailing out..." << std::endl;
```

Comment:
Is it possible that a lost ZK location gets recovered or once it's lost is it lost for good? Is there a scenario where brief network faults would cause this to be lost and the quickly recovered? I suspect it's lost for good, but I don't know for sure.